File: sorting/selection_sort.js
let arr = [
  83,
  47,
  44,
  79,
  84,
  35,
  99,
  9,
  39,
  19,
]
console.log(arr)

function swarrp(i, j) {
  let temp = arr[i]
  arr[i] = arr[j]
  arr[j] = temp
}

for (let i = 0; i < arr.length; i++) {
  let l = 999999;
  let li = i;
  for (let j = i; j < arr.length; j++) {
    if (arr[j] < l) {
      l = arr[j];
      li = j;
    }
  }
  swarrp(i, li)
  console.log(arr)
}

File: sorting/selection_sort.c
/*
        Takes O(n^2) time
*/
#include "utils.h"
#include <stdio.h>

void selection_sort(int arr[], int len) {

  for (int j = 0; j < len; j++) {
    // find smallest element in arr[j:len] and put it in
    // arr[j]
    int smallest_int = arr[j];
    int smallest_ind = j;
    for (int i = j + 1; i < len - 1; i++) {
      if (arr[i] < smallest_int) {
        smallest_int = arr[i];
        smallest_ind = i;
      }
    }
    swap(arr, j, smallest_ind);
    print_arr(arr, len);
  }
}
int main() {
  int my_arr[] = {5, 2, 4, 6, 1, 3};
  int my_arr2[] = {31, 41, 59, 26, 41, 58};
  int arr_len = 6;

  selection_sort(my_arr, arr_len);
  selection_sort(my_arr2, arr_len);
}
